discovery-provider/src/model_validator.py

```diff
@@ -0,0 +1,174 @@
+import os.path
+import json
+import copy
+import logging  # pylint: disable=C0302
+from jsonschema import Draft7Validator, ValidationError
+logger = logging.getLogger(__name__)
+
+# https://app.quicktype.io/ -- JSON schema generator
+class ModelValidator:
+    '''
+        {key: value} :
+        {
+            model (first letter uppercase): {
+                model_schema: schema,
+                field_schema: {
+                    title: schema,
+                    mood: schema,
+                    ...
+            },
+                fields: [title, mood, ...]
+            }
+        }
+    '''
+    models_to_schema_and_fields_dict = {}
+    BASE_PATH = './src/schemas/'
+    FILE_NAME = ''
+
+    @classmethod
+    def validate(cls, to_validate, model, field=""):
+        cls.FILE_NAME = model.lower() + '_schema.json'
+        try:
+            schema = cls.get_schema(field, model)
+            validator = Draft7Validator(schema)
+
+            found_invalid_field = False
+            for error in sorted(validator.iter_errors(to_validate), key=str):
+                found_invalid_field = True
+                logger.warning(f"Error with {model} instance {to_validate}: {error.message}")
+
+            # if any error occurs, raise exception
+            if found_invalid_field:
+                raise ValidationError(f"Instance {to_validate} is not proper")
+
+        except ValidationError as ve:
+            raise ve
+        except BaseException as e:
+            # one of many errors specified in helper methods
+            raise e
+
+    @classmethod
+    def get_schema(cls, field, model):
+        try:
+            # If model is not present in dict, init its schema and its field subschemas
+            if model not in cls.models_to_schema_and_fields_dict:
+                schema = cls.init_model_schemas(model)
+                if not schema:
+                    error_msg = f"ModelValidation failed. Are you sure the schema for {model} is present and proper?"
+                    raise RuntimeError(error_msg)
+
+            # If field is empty, return the entire model schema
+            if field == "":
```

Comment:
i wrote this with the intended behavior that if someone wanted to validate an entire track object, s/he does not need to specify a specific field. but, i can see how that's not clear and can be problematic.

perhaps we can make the entire model check more specific, like if someone passes `all` instead of `""`

Reply:
yeah if you make it either use like None or break it up into two functions